ground/src/main/java/com/google/android/ground/ui/common/AbstractMapContainerFragment.kt

```diff
@@ -96,14 +96,23 @@ abstract class AbstractMapContainerFragment : AbstractFragment() {
 
   private fun onCameraUpdateRequest(newPosition: CameraPosition, map: MapFragment) {
     Timber.v("Update camera: %s", newPosition)
-    if (newPosition.zoomLevel != null) {
-      var zoomLevel = newPosition.zoomLevel
-      if (!newPosition.isAllowZoomOut) {
-        zoomLevel = max(zoomLevel, map.currentZoomLevel)
-      }
-      map.moveCamera(newPosition.target, zoomLevel)
+    val bounds = newPosition.bounds
+    val target = newPosition.target
+    var zoomLevel = newPosition.zoomLevel
+
+    if (target != null && zoomLevel != null && !newPosition.isAllowZoomOut) {
+      zoomLevel = max(zoomLevel, map.currentZoomLevel)
+    }
+
+    // TODO(Shobhit): Ensure that the order of preference is correct.
```

Comment:
See comment on `CameraPosition` to refactor so that it actually represents a non-duplicated state rather than an action with sparse params.

Reply:
Added a TODO to fix this once the CameraPosition class is cleaned up